x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -293,6 +294,12 @@ public ExpressionEvaluator.Factory toEvaluator(Function<Expression, ExpressionEv
         return new ExpressionEvaluator.Factory() {
             @Override
             public ExpressionEvaluator get(DriverContext context) {
+                if (conditionsFactories.size() == 1
+                    && conditionsFactories.get(0).value.safeToEvalInLazy()
+                    && elseValueFactory.safeToEvalInLazy()) {
+
+                    return new EagerEvaluator(context, conditionsFactories.get(0).apply(context), elseValueFactory.get(context));
```

Comment:
I can't tell if the logic is wrong here or if the naming is just confusing.  To me, this reads like "if it is safe to evaluate in lazy, return an eager evaluator".  I would normally say "eager evaluation" was the opposite of "lazy evaluation".

Reply:
Bad name, I think. It's "if this is safe to eagerly evaluate when we expect a lazy"

Reply:
I'll rename this now that the CASE bugs are fixed.